### PR TITLE
docs: give better guidance about how to configure the agent TLS CA

### PR DESCRIPTION
**Pull request by dnephin**
This language is loosely based on some language from [Consul's Learn guide about enablingTLS](https://learn.hashicorp.com/tutorials/consul/tls-encryption-secure#initialize-the-built-in-ca). 

The goal is to provide some more information about the security implications of these configuration settings.

### CODE REVIEW

**Review by blake — APPROVED** [review]
LGTM

File: website/content/docs/agent/options.mdx [hunk 1]
```diff
@@ -2211,6 +2211,12 @@ This section documents all of the configuration settings that apply to Agent TLS
 TLS is used by the HTTP API, server RPC, and xDS interfaces. Some of these settings may also be
 applied automatically by [auto_config](#auto_config) or [auto_encrypt](#auto_encrypt).
 
+~> **Security Note:** The Certificate Authority (CA) specified by `ca_file` and `ca_path`
+should use a private CA, not a public one.  We also recommend using a separate CA for
+Consul and not sharing the CA with any other systems. Any certificate signed by the
+CA will be allowed to communicate with the cluster and a specially crafted certificate
+signed by the CA can gain full read and write access to Consul.
```
Comment (picatz) [inline website/content/docs/agent/options.mdx:2218]: This is a great addition to the docs!

Small wording suggestion, what do you think?

```suggestion
~> **Security Note:** The Certificate Authority (CA) specified by `ca_file` or `ca_path`
should be a private CA, not a public one. We recommend using a dedicated CA
which should not be used with any other systems. Any certificate signed by the
CA will be allowed to communicate with the cluster and a specially crafted certificate
signed by the CA can be used to gain full access to Consul.
```

Comment (dnephin) [reply website/content/docs/agent/options.mdx:2218]: Sounds great, thank you!

**Comment by hc-github-team-consul-core** [comment]
:cherries: If backport labels were added before merging, cherry-picking will start automatically.

To retroactively trigger a backport after merging, add backport labels and re-run https://circleci.com/gh/hashicorp/consul/418407.

**Comment by hc-github-team-consul-core** [comment]
:cherries: If backport labels were added before merging, cherry-picking will start automatically.

To retroactively trigger a backport after merging, add backport labels and re-run https://circleci.com/gh/hashicorp/consul/418423.

**Comment by hc-github-team-consul-core** [comment]
:cherries: If backport labels were added before merging, cherry-picking will start automatically.

To retroactively trigger a backport after merging, add backport labels and re-run https://circleci.com/gh/hashicorp/consul/418475.

**Comment by hc-github-team-consul-core** [comment]
:cherries: If backport labels were added before merging, cherry-picking will start automatically.

To retroactively trigger a backport after merging, add backport labels and re-run https://circleci.com/gh/hashicorp/consul/418475.

**Comment by hc-github-team-consul-core** [comment]
:cherries::white_check_mark: Cherry pick of commit 051f42085516d95591a882f1fcda1a072e478d47 onto `stable-website` succeeded!

**Comment by hc-github-team-consul-core** [comment]
:cherries::white_check_mark: Cherry pick of commit 051f42085516d95591a882f1fcda1a072e478d47 onto `release/1.10.x` succeeded!

**Pull request merged**